### PR TITLE
Added new redis.instance resource attribute and config parameter

### PR DESCRIPTION
**Pull request by manang-splunk**
**Description:** Added new redis.instance resource attribute and also added config parameter redis_instance to make it configurable.

**Testing:** Updated the existing test case to verify redis.instance attribute.

**Documentation:** Added new config parameter in readme.md file.

### CODE REVIEW

File: receiver/redisreceiver/redis_scraper_test.go [hunk 1]
```diff
@@ -28,13 +28,15 @@ func TestRedisRunnable(t *testing.T) {
...
 	settings.Logger = logger
-	runner, err := newRedisScraper(newFakeClient(), settings)
+	runner, err := newRedisScraper(newFakeClient(), settings, "TestEndpoint")
 	require.NoError(t, err)
 	md, err := runner.Scrape(context.Background())
 	require.NoError(t, err)
 	// + 6 because there are two keyspace entries each of which has three metrics
 	assert.Equal(t, len(getDefaultRedisMetrics())+6, md.DataPointCount())
 	rm := md.ResourceMetrics().At(0)
+	value, _ := rm.Resource().Attributes().Get("redis.instance")
```
[the remark this answers is not in the document]
Comment (manang-splunk) [reply receiver/redisreceiver/redis_scraper_test.go:39]: As `redis.instance` is attribute specific to redis receiver so we do not need to add it to convention.

**Review by rmfitzpatrick — APPROVED** [review]

File: receiver/redisreceiver/config.go [hunk 2]
```diff
@@ -33,4 +33,6 @@ type Config struct {
 	Password string `mapstructure:"password"`
 
 	TLS configtls.TLSClientSetting `mapstructure:"tls,omitempty"`
+	//Optional redis instance. Name of the redis instance
```
Comment (rmfitzpatrick) [inline receiver/redisreceiver/config.go:36]:
```suggestion
	// Optional redis instance for identifying resource attribute. Can be used to set the reported name of the redis instance, with a default of the configured `Endpoint`.
```

File: receiver/redisreceiver/redis_scraper_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply receiver/redisreceiver/redis_scraper_test.go:39]: That is not true :) conventions are even for a specific product/database if they are intended to be transferred on the wire for example between this collector and a backend (splunk for example). It is still a convention, and without having that in the specs it looks like a vendor specific convention.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
See comment about `redis_instance`

**Pull request closed without merging**